app/controllers/v1/profiles_controller.rb

```diff
@@ -4,15 +4,12 @@ module V1
   # API for Profiles
   class ProfilesController < ApplicationController
     include InventoryServiceHelper
+    include ProfileAttributes
 
-    ALLOWED_CREATE_ATTRIBUTES = %i[
-      name description compliance_threshold business_objective
-      parent_profile_id
-    ].freeze
-
-    ALLOWED_UPDATE_ATTRIBUTES = %i[
-      name description compliance_threshold business_objective
-    ].freeze
+    before_action only: %i[update] do
+      error = 'Editing an external profile is forbidden.'
+      render json: { errors: error }, status: :forbidden if profile.policy.nil?
```

Comment:
Observation: There could be a common method for rendering an error. The existing `render_error` can also accept String (or an array of Strings).